src/commands/serve.rs

```diff
@@ -432,7 +434,28 @@ impl ServeCommand {
                     .serve_connection(
                         stream,
                         hyper::service::service_fn(move |req| {
-                            handle_request(h.clone(), req, comp.clone())
+                            let comp = comp.clone();
+                            let h = h.clone();
+                            async move {
+                                use http_body_util::{BodyExt, Full};
+                                fn to_errorcode(_: Infallible) -> ErrorCode {
+                                    unreachable!()
+                                }
+                                match handle_request(h, req, comp).await {
+                                    Ok(r) => Ok::<_, Infallible>(r),
+                                    Err(e) => {
+                                        eprintln!("error: {e:?}");
+                                        Ok(Response::builder()
+                                            .status(StatusCode::INTERNAL_SERVER_ERROR)
+                                            .body(
+                                                Full::new(bytes::Bytes::new())
```

Comment:
Shouldn't this at show a basic html page with a user visible message that an internal server error happened like Apache and Nginx do? A completely empty page is even more confusing for end users than the browser telling the user that the connection got closed.

Reply:
Opened https://github.com/bytecodealliance/wasmtime/pull/10851